components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/WebsocketInboundHandler.java

```diff
@@ -223,10 +222,19 @@ private boolean validateOAuthHeader(FullHttpRequest req) throws APISecurityExcep
             version = getVersionFromUrl(uri);
             APIKeyValidationInfoDTO info;
             if (!req.headers().contains(HttpHeaders.AUTHORIZATION)) {
-                log.error("No Authorization Header Present");
-                return false;
+                QueryStringDecoder decoder = new QueryStringDecoder(req.getUri());
+                Map<String, List<String>> requestMap = decoder.parameters();
+                if (requestMap.containsKey(APIConstants.AUTHORIZATION_QUERY_PARAM_DEFAULT)) {
+                    req.headers().add(HttpHeaders.AUTHORIZATION, APIConstants.CONSUMER_KEY_SEGMENT + ' '
+                                    + requestMap.get(APIConstants.AUTHORIZATION_QUERY_PARAM_DEFAULT).get(0));
```

Comment:
Shall we remove the the query param from the URI here, or else the backend also will receive the token.

Reply:
done